isis/tests/FunctionalTestsCaminfo.cpp

```diff
@@ -3,122 +3,311 @@
+        PvlKeyword currentKey = parameters[i];
+        CSVReader::CSVAxis currentColumn = caminfoCsv.getColumn(currentKey.name());
+
+        // Skip RunDate as it's not in the CSV
+        if (currentKey.name() != "RunDate") {
+            EXPECT_TRUE(currentColumn.dim1() > 0) << "Failed to find column [" <<
+                    currentKey.name().toStdString() << "] in CSV";
+            QString stringCsvValue = currentColumn[0];
+            bool isCsvNumeric = false;
+            double numericCsvValue = stringCsvValue.toDouble(&isCsvNumeric);
+            QString stringPvlValue = QString(currentKey);
+            bool isPvlNumeric = false;
+            double numericPvlValue = stringPvlValue.toDouble(&isPvlNumeric);
+            if (isCsvNumeric && isPvlNumeric) {
+                EXPECT_NEAR(numericCsvValue, numericPvlValue, 0.001) <<
+                        "Column [" << currentKey.name().toStdString() << "] value [" <<
+                        stringCsvValue.toStdString() << "] does not match Pvl value [" <<
+                        stringPvlValue.toStdString() << "] within tolerance [0.001].";
+            }
+            else {
+                EXPECT_EQ(stringCsvValue.toStdString(), stringPvlValue.toStdString()) <<
+                        "Column [" << currentKey.name().toStdString() << "] value [" <<
+                        stringCsvValue.toStdString() << "] does not match Pvl value [" <<
+                        stringPvlValue.toStdString() << "].";
+            }
+        }
+    }
+
+    for (int i = 0; i < camstats.keywords(); i++) {
+        PvlKeyword currentKey = camstats[i];
+        QString columnName = "CamStats_" + currentKey.name();
+        CSVReader::CSVAxis currentColumn = caminfoCsv.getColumn(columnName);
+        EXPECT_TRUE(currentColumn.dim1() > 0) << "Failed to find column [" <<
+                                                 columnName.toStdString() << "] in CSV";
+        QString stringCsvValue = currentColumn[0];
+        bool isCsvNumeric = false;
+        double numericCsvValue = stringCsvValue.toDouble(&isCsvNumeric);
+        QString stringPvlValue = QString(currentKey);
+        bool isPvlNumeric = false;
+        double numericPvlValue = stringPvlValue.toDouble(&isPvlNumeric);
+        if (isCsvNumeric && isPvlNumeric) {
+            EXPECT_NEAR(numericCsvValue, numericPvlValue, 0.001) <<
+                    "Column [" << columnName.toStdString() << "] value [" <<
+                    stringCsvValue.toStdString() << "] does not match Pvl value [" <<
+                    stringPvlValue.toStdString() << "] within tolerance [0.001].";
+        }
+        else {
+            EXPECT_EQ(stringCsvValue.toStdString(), stringPvlValue.toStdString()) <<
+                    "Column [" << columnName.toStdString() << "] value [" <<
+                    stringCsvValue.toStdString() << "] does not match Pvl value [" <<
+                    stringPvlValue.toStdString() << "].";
+        }
```

Comment:
I considered writing a new predicate formatter for this comparison case as it comes up a lot but I don't know if it'll be worth it. The predicate formatters are kind of clunky to use.